panc/src/main/java/org/quattor/pan/dml/data/ProtectedListResource.java

```diff
@@ -39,13 +43,28 @@ public Element put(int index, Element newValue) {
 	}
 
 	@Override
+	public void append(Element e) {
+		throw CompilerError.create(MSG_ILLEGAL_WRITE_TO_PROTECTED_LIST);
+	}
+
+	@Override
+	public void prepend(Element e) {
+		throw CompilerError.create(MSG_ILLEGAL_WRITE_TO_PROTECTED_LIST);
+	}
+
+	@Override
 	public int size() {
 		return baseList.size();
 	}
 
 	@Override
 	public Resource.Iterator iterator() {
-		return baseList.iterator();
+		return baseList.protectedIterator();
```

Comment:
i think what you need here instead of the `protectedIterator` is 
```
baseList = baseList.protect();
return baseList.iterator();
```

(well, but then code that works :smile: )